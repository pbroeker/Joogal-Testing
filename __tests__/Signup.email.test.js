import React from 'react';
import Email from '../Signup/Signup.Email';
import { fireEvent, render } from '@testing-library/react-native';
import { Alert } from 'react-native';

jest.spyOn(Alert, 'alert').mockImplementation(() => {});

describe('Signup Email', () => {
  const setSignup = jest.fn();
  const navigation = {navigate : jest.fn()};
  let renderedComponent;

  beforeEach(() => {
    Alert.alert.mockReset();
    renderedComponent = render( <Email navigation={navigation} setSignup={setSignup} /> );
  })

  test('snapshot', () => {
    const { toJSON } = renderedComponent;
    expect(toJSON()).toMatchSnapshot();
  })

  test('renders correctly', () => {
    const { getByText } = renderedComponent;
    expect(getByText('What\'s your email address?')).toBeTruthy();
  })

  test('calls signup correctly', () => {
    const { getByText, getByPlaceholderText } = renderedComponent;
    fireEvent(getByPlaceholderText('your@email.com*'), 'onChangeText', 'workingMailAddress@gmail.com');
    fireEvent(getByText('➤'), 'onPress' );
    expect(setSignup).toHaveBeenCalledTimes(1);
  })

  test('gives an alert on too short mailadress', () => {
    const { getByPlaceholderText, getByText } = renderedComponent;
    fireEvent(getByPlaceholderText('your@email.com*'), 'onChangeText', 'a@b');
    fireEvent(getByText('➤'), 'onPress' );

    expect(Alert.alert).toHaveBeenCalledWith('Not a valid email address');
    expect(Alert.alert).toHaveBeenCalledTimes(1);
  })

  test('gives an alert on mailadress without @', () => {
    const { getByPlaceholderText, getByText } = renderedComponent;
    fireEvent(getByPlaceholderText('your@email.com*'), 'onChangeText', 'mailAddressWithout');
    fireEvent(getByText('➤'), 'onPress' );

    expect(Alert.alert).toHaveBeenCalledWith('Not a valid email address');
    expect(Alert.alert).toHaveBeenCalledTimes(1);
  })
})