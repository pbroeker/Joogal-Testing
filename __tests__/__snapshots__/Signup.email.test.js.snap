// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Email snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1d1e20",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "20%",
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 30,
          "marginBottom": 30,
          "marginLeft": "10%",
        }
      }
    >
      What's your email address?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "45%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="email"
      autoFocus={true}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="your@email.com*"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "whitesmoke",
          "marginVertical": "1%",
          "padding": 14,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginTop": 100,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "teal",
          "borderRadius": 50,
          "marginRight": 30,
          "opacity": 1,
          "padding": 10,
          "width": 56,
        }
      }
      testID="emailContinueButton"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "600",
          }
        }
      >
        âž¤
      </Text>
    </View>
  </View>
</View>
`;
